### ISSUE TITLE
Add multi-file build feature

### ISSUE THREAD

**Issue by Jagerente** [Jagerente/aottg2cl-vsce#60]
## Info

Enhance the VSCode extension to support annotation tags for imports and implement a build command that generates a final script file based on these annotations.

### Add support for the following annotation tags. Multiple import per line:
```cs
# @import FileA FileB
# @import FileC

class Main {  /* */ }
```
Tags should be declared at the very beginning of the file.

### Build Command

Add a new command accessible via `Ctrl + Shift + P` (Show All Commands)

When executed, this command should:
1. Ask the user for the name of the final file.
  - Default name: `<current_file_name>_final`
2. Analyze current file for imports
  - Parse all imports specified by annotation tags
  - For each imported file, recursively parse its imports
  - Build the dependency chain
3. Validate and optimize the import process
  - Ensure:
    - No cyclic dependencies: If detected, throw an error
    - No duplicate classes: If detetcted, throw an error.
    - Files imported by multiple files are deduplicated and included only once in the final build.
4. Generate the final file:
  - Combine the current file and all its imports sequentially
  - Append imports to the end of the parent file in the correct order.
  - Save the generated file in the same directory with a user provided name in the beginning.

#### Files 

- FileA.acl
```cs
# @import ClassC

class ClassA { }

class ClassB { }
```

- FileB.acl
```cs
class ClassC { }
```

- Main.acl
```cs
# @import ClassA

class Main { /* */ }
```

#### Expected behaviour:
1. The user runs the Build command on Main.acl.
2. The generated file `Main_final.acl` should contain:

```cs
class Main { /* */ }

class ClassC { }

class ClassA { }

class ClassB { }
```

#### Files

- FileA.acl
```cs
# @import ClassC

class ClassA { }

class ClassB { }
```

- FileB.acl
```cs
class ClassB { } # Duplicate of ClassB in FileA
```

- Main.acl
```cs
# @import ClassA ClassB

class Main { /* */ }
```

#### Expected behaviour:
1. The user runs the Build command on Main.acl.
2. The command detects a duplicate class definition: ClassB in FileA.acl and FileB.acl.
3. The command stops with an error: `Error: Duplicate class definition detected: ClassB found in FileA.acl and FileB.acl.`

#### Files

- FileA.acl
```cs
# @import ClassC

class ClassA { }
```

- FileB.acl
```cs
# @import ClassA

class ClassC { }
```

- Main.acl
```cs
# @import ClassA

class Main { /* */ }
```

#### Expected behaviour:
1. The user runs the Build command on Main.acl.
2. The command detects a cyclic dependency: FileA.acl -> FileB.acl -> FileA.acl
3. The command stops with an error: `Error: Cyclic dependency detected: FileA.acl -> FileB.acl -> FileA.acl.`

#### Files

- FileA.acl
```cs
# @import ClassC

class ClassA { }
```

- FileB.acl
```cs
# @import ClassC

class ClassB { }
```

- FileC.acl
```cs
class ClassC { }
```

- Main.acl
```cs
# @import ClassA ClassB

class Main { /* */ }
```

#### Expected behaviour:
1. The user runs the Build command on Main.acl.
2. The command deduplicates ClassC, as it is imported by both FileA.acl and FileB.acl
3. The generated file `Main_final.acl` should contain:
```cs
class Main { /* */ }

class ClassC { }

class ClassA { }

class ClassB { }
```
